Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -385,6 +385,31 @@ void ACesium3DTileset::SetTilesetSource(ETilesetSource InSource) {
   }
 }
 
+bool CompareMaps(
+    const TMap<FString, FString>& Lhs,
+    const TMap<FString, FString>& Rhs) {
+  if (Lhs.Num() != Rhs.Num()) {
+    return false;
+  }
+
+  for (auto& [Key, Value] : Lhs) {
+    const FString* RhsVal = Rhs.Find(Key);
+    if (!RhsVal || *RhsVal != Value) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void ACesium3DTileset::SetRequestHeaders(
+    const TMap<FString, FString>& InRequestHeaders) {
+  if (!CompareMaps(InRequestHeaders, this->RequestHeaders)) {
+    this->DestroyTileset();
```

Comment:
This handles the C++ and Blueprint cases, but when headers are changed on the Details panel, the tileset won't refresh. To fix that, add this new property to `PostEditChangeProperty`.